plugins/passwordagentauthplugin/passwordagentplugin.cpp

```diff
@@ -523,9 +526,15 @@ Result PasswordAgentPlugin::beginAuthentication(
                     .compare(QStringLiteral("true"), Qt::CaseInsensitive) == 0) {
                 result = Result(Result::InteractionViewUserCanceledError,
                                 QLatin1String("The user canceled the authentication dialog"));
-            } else {
+            } else if (!QDBusReply<PolkitAuthorizationResult>(*watcher).value().details.keys().isEmpty()) {
                 result = Result(Result::IncorrectAuthenticationCodeError,
                                 QStringLiteral("Password Agent was unable to verify the authenticity of the user"));
+            } else {
+                // The calling application was not started by the lipstick user session
+                // so polkit cannot work for it (as the lipstick-security-ui is for that session).
+                // Instead, use devicelock API directly.
```

Comment:
Besides the comment,  I'd suggest to add a `qcDebug` message here. That would help if this comes up during debugging.

Reply:
Good point, I will add this one.